x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/index_migrator/index.test.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import { RuleMigrationIndexMigrator } from '.';
+import * as RuleMigrationSpaceIndexMigratorModule from './per_space';
+import type { Adapters } from '../types';
+
+const esClientMock = {
+  indices: {
+    get: jest.fn().mockResolvedValue({
+      '.kibana-siem-rule-migrations-rules-space1': {},
+      '.kibana-siem-rule-migrations-rules-space2': {},
+      '.kibana-siem-rule-migrations-rules-space3': {},
+    }),
+  },
+} as unknown as ElasticsearchClient;
+
+const ruleMigrationIndexAdapters = {
+  rules: { name: '.kibana-siem-rule-migrations-rules-*' },
```

Comment:
This is not correct, adapters `name` are just the prefixes, they never include the `*`, so this should be tested with `.kibana-siem-rule-migrations-rules`.

reference: 
https://github.com/elastic/kibana/blob/58ba61bdd194a8c8fe7f49ebbecdf78ba7e2da4c/x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_service.ts#L73-L75

Reply:
Let's use the actual IndexPatternAdapter (without installing anything)

```ts
const rulesIndexName = '.kibana-siem-rule-migrations-rules';
const esClientMock = {
  indices: {
    get: jest.fn().mockResolvedValue({
      [`${rulesIndexName}-space1`]: {},
      [`${rulesIndexName}-space2`]: {},
      [`${rulesIndexName}-space3`]: {},
    }),
  },
} as unknown as ElasticsearchClient;

const ruleMigrationIndexAdapters = {
  rules: new IndexPatternAdapter(rulesIndexName, {
    kibanaVersion: '9.0.0',
  }),
} as unknown as Adapters;
```

Reply:
Actually, I can confirm that `*` is included in the returned pattern. I had tried upgrade multiple times and it was able to migration without any issues.

![image](https://github.com/user-attachments/assets/603455b0-4cff-49b8-b6a9-cd79e8ba1b62)

Reference: 

https://github.com/elastic/kibana/blob/666b466bf315c6beafabb57975418adcecaf7aad/x-pack/solutions/security/packages/index-adapter/src/index_pattern_adapter.ts#L17-L19

Reply:
You are right! My bad, your approach is correct. The `name` of `IndexPatternAdapter` includes `-*`, I was confused with the `prefix` attribute, which does not include it. 

Ok, your code works. However, it relies on an internal implementation detail of the `IndexPatternAdapter`. I would still adapt this code to make the logic intentional, and append `*` explicitly for the ES request. 
This test as well.